tiatoolbox/tools/registration/wsi_registration.py

```diff
@@ -0,0 +1,33 @@
+import numpy as np
+from skimage import exposure, filters, morphology
+
+
+def preprocess(fixed_img, moving_img):
+    """This function performs normalization to unify the appearance of fixed
+     and moving images
+
+    Args:
+            fixed_img (:class:`numpy.ndarray`):
+                    A grayscale fixed image
+            moving_img (:class:`numpy.ndarray`):
+                    A grayscale moving image
+
+    Returns:
+            :class:`numpy.ndarray`:
+                    A normalized grayscale fixed image
+            :class:`numpy.ndarray`:
+                    A normalized grayscale moving image
+
+    """
+    if len(fixed_img.shape) != 2 or len(moving_img.shape) != 2:
```

Comment:
You can have an image of shape (2, 2, 1) where this would fail.

One way could be to apply np.squeeze before checking length. Another way would be to check shape[2] == 1.